src/properties.ts

```diff
@@ -135,11 +135,15 @@ function dependenceExists(pkg: PackageJson, pkgName: string): boolean {
 
 function populatePackageParams(params: { [key: string]: string | {} }, pkg: PackageJson) {
   const invalidCharacterRegex = /[?$*+~.()'"!:@/]/g;
-  params['sonar.projectKey'] = slugify(pkg.name, {
-    remove: invalidCharacterRegex,
-  });
-  params['sonar.projectName'] = pkg.name;
-  params['sonar.projectVersion'] = pkg.version;
+  if (pkg.name) {
+    params['sonar.projectKey'] = slugify(pkg.name, {
+      remove: invalidCharacterRegex,
+    });
+    params['sonar.projectName'] = pkg.name;
+  }
+  if (pkg.version) {
+    params['sonar.projectVersion'] = pkg.version;
+  }
```

Comment:
I've made the change for `version` as well, to avoid having `params['sonar.projectVersion'] = undefined` unnecessarily